### PR TITLE
Add ROADMAP.md 🤖

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This is extracted from the [Tekton CLI 2020 Roadmap](https://docs.google.com/document/d/1NfvmJk2uQf7vYshrkpQ8FG98TLQlnfSmvQkuQC0R1VA/edit#).

/area roadmap

/cc @chmouel @piyush-garg @danielhelfand @sbwsg 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+### V1beta1 support and GA
+
+- Add support for v1beta1 and later on GA
+- Warn usage of PipelineResource in v1beta1 (no error but warn that it is not supported aka will work but can be broken in a later release)
+- Support for workspace, result, …
+
+### Improve e2e Testing Coverage for CLI
+
+- Get #505 merged to convert e2e testing to Go
```
Comment (danielhelfand) [inline ROADMAP.md]: Maybe link to #505 here?

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- There are two issues open around this:
+  - https://github.com/tektoncd/cli/issues/574
+  - https://github.com/tektoncd/cli/issues/575
+- There have been requests for a create command for the CLI similar to kubectl create/apply
+- We should support one of the above strategies, or we will not support creation/updating like kubectl create/apply -f moving forward
+- If we do not support a -f option, we should remove the create -f commands from pipeline/task/resource
+
+### Catalog integration
+
+- Related to the OCI experiment (#2137) and the catalog / hub, integrate better with the/a catalog
```
Comment (danielhelfand) [inline ROADMAP.md]: Link to https://github.com/tektoncd/pipeline/pull/2137

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  - Allow searching catalog for a Task, Pipeline, …
+  - Allow importing a task from the catalog into the cluster
+  - … ?
```
Comment (danielhelfand) [inline ROADMAP.md]: We can remove last point

File: ROADMAP.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+### Local to cluster build
+
+- From a user working directory, be able to run a task or pipeline with the content of this working directory
+  - Use volume (pvc, …) with `workspace` support
+  - Create / Update volume and start the pipeline
+- Related to :  Binary (local) input type #924
```
Comment (danielhelfand) [inline ROADMAP.md]: Link to https://github.com/tektoncd/pipeline/issues/924

**Comment by chmouel** [comment]
I don't think there is much more to add on what you wrote...

I wanted to add something I talk about before but never took the time to write it in a issue. 

I want to be able from the cli auto generate some Pipeline/tasks from a source code folder detecting what kind of source code it is (i.e: go, java, python, ruby, rust, project etc...) and automatically apply  the standard task from the catalog.

The detection and generation is not that hard but to know what kind of task that needs to be used may involve some improvement to the OCI catalog work first. so let's wait for that work to land before coming back to this...

On CI, it seems that make golden has some issue...

**Comment by vdemeester** [comment]
@danielhelfand @chmouel updated :wink:

**Pull request merged**